src/main/java/ch/njol/skript/expressions/base/EventValueExpression.java

```diff
@@ -223,16 +233,33 @@ private <E extends Event> T getValue(E event) {
 	@Nullable
 	@SuppressWarnings("unchecked")
 	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.SET && !eventConverters.isEmpty()) {
+			if (isDelayed) {
+				Skript.error("Event values cannot be changed after the event has already passed.");
+				return null;
+			}
+			return CollectionUtils.array(type);
+		}
 		if (changer == null)
 			changer = (Changer<? super T>) Classes.getSuperClassInfo(componentType).getChanger();
 		return changer == null ? null : changer.acceptChange(mode);
 	}
 
 	@Override
 	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
-		if (changer == null)
-			throw new SkriptAPIException("The changer cannot be null");
-		ChangerUtils.change(changer, getArray(event), delta, mode);
+		if (mode == ChangeMode.SET) {
+			EventConverter<Event, T> converter = eventConverters.get(event.getClass());
+			if (converter != null) {
+				if (!type.isArray() && delta != null) {
+					converter.set(event, (T)delta[0]);
+				} else {
+					converter.set(event, (T)delta);
+				}
+			}
+		}
+		if (changer != null) {
+			ChangerUtils.change(changer, getArray(event), delta, mode);
```

Comment:
Do you think it's safe to have both of these run at the same time? It's probably rare but in case the classinfo has its own changer that allows set, I think it'd be best to only let one or the other run.

Reply:
You're right. I think I mistakenly was thinking about the `acceptChange` not handling the changer if this converter is accepted, and just mistakenly assuming it would get ignored.